docs/hub/webhooks-guide-auto-retrain.md

```diff
@@ -0,0 +1,176 @@
+# Webhook guide: Setup an automatic system to re-train a model when a dataset changes
+
+This guide will guide you through the set up of an automatic training pipeline on the HuggingFace platform,
+using HF Webhooks, Spaces and AutoTrain.
+
+We will build a Webhook that listen to changes on an image classification dataset, and triggers a fine-tuning
+of [microsoft/resnet-50](https://huggingface.co/microsoft/resnet-50) using [AutoTrain](https://huggingface.co/autotrain).
+
+
+
+## Pre-requisite: upload your dataset to the Hub
+
+We will use a [simple image classification dataset](https://huggingface.co/datasets/sbrandeis-test-org/input-dataset) for the sake
+of the example. Learn more about uploading your data to the Hub here: https://huggingface.co/docs/datasets/upload_dataset.
+
+![dataset](../assets/auto-retrain/dataset.png)
+
+## Create a webhook to react to the dataset's changes
+
+First, let's create a webhook from your settings: https://huggingface.co/settings/webhooks.
+
+- Select your dataset as the target repository. We will target [sbrandeis-test-org/input-dataset](https://huggingface.co/datasets/sbrandeis-test-org/input-dataset) in this example.
+- You can put a dummy Webhook URL for now. Defining your webhook will let you take a look at the events that are going to be sent to it (and you can replay them, which will be useful for debugging).
+- Input a secret as it will be more secure.
+- Subscribe to "Repo update" events as we want to react to data changes
+
+Your webhook will look like this:
+
+![webhook-creation](../assets/auto-retrain/webhook-creation.png)
+
+## Create a Space to react to your webhook
+
+We now need a way to react to your webhook events. An easy way is to use a [Space](https://huggingface.co/docs/hub/spaces-overview)!
+
+You can find an example Space here: https://huggingface.co/spaces/sbrandeis-test-org/active-learning-webhook/tree/main.
+
+This Space uses Docker, Python, [FastAPI](https://fastapi.tiangolo.com/) and [uvicorn](https://www.uvicorn.org) to run a simple HTTP server. Read more about Docker Spaces [here](https://huggingface.co/docs/hub/spaces-sdks-docker).
+
+
+The entry point is [src/main.py](https://huggingface.co/spaces/sbrandeis-test-org/auto-retrain/blob/main/src/main.py). Let's walk through this file and details what it does:
+
+1. It spawns a FastAPI app that will listen to HTTP `POST` requests on `/webhook`:
+
+```python
+from fastapi import FastAPI
+
+# [...]
+@app.post("/webhook)
+async def post_webhook(
+	# ...
+):
+	# ...
+```
+
+2. This route checks that the `X-Webhook-Secret` header is present, and that its value is the same as the one you set in your webhook's settings:
+
+```python
+# [...]
+
+WEBHOOK_SECRET = os.getenv("WEBHOOK_SECRET")
```

Comment:
Does this have to be the same secret defined in repository? Maybe note if so 🙂

Reply:
Good point - I added a note about that!